src/Mod/Sketcher/Gui/EditModeGridCoinManager.cpp

```diff
@@ -0,0 +1,282 @@
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+
+#include "PreCompiled.h"
+
+#ifndef _PreComp_
+# include <Inventor/nodes/SoSeparator.h>
+# include <Inventor/nodes/SoGroup.h>
+# include <Inventor/nodes/SoSwitch.h>
+# include <Inventor/nodes/SoMaterial.h>
+# include <Inventor/nodes/SoCoordinate3.h>
+# include <Inventor/nodes/SoLineSet.h>
+# include <Inventor/nodes/SoFont.h>
+
+# include <Inventor/nodes/SoMarkerSet.h>
+# include <Inventor/nodes/SoTranslation.h>
+# include <Inventor/nodes/SoText2.h>
+# include <Inventor/nodes/SoPickStyle.h>
+# include <Inventor/nodes/SoDrawStyle.h>
+# include <Inventor/SoPickedPoint.h>
+# include <Inventor/details/SoPointDetail.h>
+# include <Inventor/details/SoDetail.h>
+# include <Inventor/details/SoLineDetail.h>
+
+# include <Inventor/nodes/SoAnnotation.h>
+# include <Inventor/nodes/SoImage.h>
+# include <Inventor/nodes/SoInfo.h>
+
+# include <Inventor/actions/SoRayPickAction.h>
+
+# include <Inventor/SbVec3f.h>
+# include <Inventor/SbImage.h>
+
+# include <memory>
+#endif  // #ifndef _PreComp_
+
+#include <Gui/Inventor/SmSwitchboard.h>
+
+#include <Base/Exception.h>
+#include <Base/Tools2D.h>
+#include <Base/UnitsApi.h>
+#include <Gui/Application.h>
+#include <Gui/Utilities.h>
+#include <Gui/View3DInventor.h>
+#include <Gui/View3DInventorViewer.h>
+#include <Base/Converter.h>
+#include <Base/Tools.h>
+
+#include <Base/Vector3D.h>
+
+#include <App/ObjectIdentifier.h>
+
+#include <Gui/SoFCBoundingBox.h>
+#include <Gui/Tools.h>
+
+#include <qpainter.h>
+
+#include "ViewProviderSketch.h"
+
+#include "ViewProviderSketchCoinAttorney.h"
+
+#include "EditModeCoinManager.h"
+
+#include "EditModeGridCoinManager.h"
+
+using namespace SketcherGui;
+using namespace Sketcher;
+
+//**************************** EditModeGridCoinManager class ******************************
+
+EditModeGridCoinManager::EditModeGridCoinManager(   ViewProviderSketch &vp,
+                                                    EditModeScenegraphNodes & editModeScenegraph):
+    viewProvider(vp),
+    editModeScenegraphNodes(editModeScenegraph),

```

Comment:
I have a warning here saying that "initializer order doesn't match the declaration order"